### PR TITLE
#3 extend fishherman registry configuration

### PR DESCRIPTION
**Pull request by KyryloShatskyiAcoustic**

### CODE REVIEW

File: fisherman-registry/src/test/java/fishermanregistry/collector/CollectorControllerIntegrationTest.java [hunk 1]
```diff
@@ -39,6 +42,9 @@ public class CollectorControllerIntegrationTest {
     @Autowired
     private CollectorRepository collectorRepository;
 
+    @Autowired
+    private ToolSource toolSource;
```
Comment (piotrwestphal) [inline fisherman-registry/src/test/java/fishermanregistry/collector/CollectorControllerIntegrationTest.java:46]: Don't use `toolSource` in integration tests. Prepare properties for int tests and check the specific values in tests. You can put these values in test resources in the `application.yml` file.

File: fisherman-registry/src/test/java/fishermanregistry/collector/tool/ToolPropertiesTest.java [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.assertj.core.api.Assertions.assertThatThrownBy;
+
+@SpringBootTest
+@ActiveProfiles("test")
+public class ToolPropertiesTest {
+
+    @Autowired
+    private ToolSource toolSource;
```
Comment (piotrwestphal) [inline fisherman-registry/src/test/java/fishermanregistry/collector/tool/ToolPropertiesTest.java:17]: As above. Read values from a properties file prepared only for tests.

File: fisherman-registry/src/test/java/fishermanregistry/fishsource/RestFishSourceTest.java [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+
+    @Autowired
+    private RestFishSource restFishSource;
+    @Autowired
+    private FishSourceProperties fishSourceProperties;
+
+    @Test
+    public void fishSourceShouldCollectFish() {
+        stubFor(post(fishSourceProperties.getPath())
+                .willReturn(ok().withBody("[{\"size\" : 1}]")));
```
Comment (piotrwestphal) [inline fisherman-registry/src/test/java/fishermanregistry/fishsource/RestFishSourceTest.java:29]: Read json from a file, and add some more fish to it :).

File: fisherman-registry/src/test/java/fishermanregistry/collector/tool/ToolPropertiesTest.java [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+    @ParameterizedTest
+    @ValueSource(strings = {"fishing-rod", "fish-net", "fishing-boat", "fishing-vessel"})
+    public void toolPropertiesLoaded(String toolName) {
+        assertThat(toolSource.getTool(toolName))
+                .extracting("name")
+                .isEqualTo(toolName);
+    }
+
+    @ParameterizedTest
+    @ValueSource(strings = {"Nonexistent Tool"})
```
Comment (piotrwestphal) [inline fisherman-registry/src/test/java/fishermanregistry/collector/tool/ToolPropertiesTest.java:28]: `@ParameterizedTest` - not needed here

File: fisherman-registry/src/test/java/fishermanregistry/DatasourceProxy.java [hunk 5]
```diff
@@ -0,0 +1,59 @@
...
+import org.springframework.beans.factory.config.BeanPostProcessor;
+import org.springframework.context.annotation.Profile;
+import org.springframework.stereotype.Component;
+import org.springframework.util.ReflectionUtils;
+import javax.sql.DataSource;
+import java.lang.reflect.Method;
+
+@Component
+@Profile("test")
+public class DatasourceProxy implements BeanPostProcessor {
```
[the remark this answers is not in the document]
Comment (KyryloShatskyiAcoustic) [reply fisherman-registry/src/test/java/fishermanregistry/DatasourceProxy.java:17]: It prints more detailed output about queries in hibernate. I'm using it to look at batches and how they are executing, it is the only way to track batches in hibernate, it seems

File: fisherman-registry/src/main/java/fishermanregistry/fishsource/rest/RestFishSource.java [hunk 6]
```diff
@@ -0,0 +1,45 @@
...
+import java.util.List;
+
+@Service
+@RequiredArgsConstructor
+@Slf4j
+public class RestFishSource implements FishSource {
+    @Qualifier("fishSourceRestTemplate")
+    private final RestTemplate restTemplate;
+    @Qualifier("listOfFishCollectionType")
+    private final CollectionType listOfFishType;
```
[the remark this answers is not in the document]
Comment (KyryloShatskyiAcoustic) [reply fisherman-registry/src/main/java/fishermanregistry/fishsource/rest/RestFishSource.java:25]: So, I can just move the initiation from beans to "@PostConstruct"?

[the remark this answers is not in the document]
Comment (piotrwestphal) [reply fisherman-registry/src/main/java/fishermanregistry/fishsource/rest/RestFishSource.java:25]: I am not sure what is the purpose of such a parameter. Maybe add `FishesResponse` first and then we will decide if it is still needed,

File: fisherman-registry/src/test/java/fishermanregistry/collector/CollectorControllerIntegrationTest.java (hunk 1 above)
Comment (KyryloShatskyiAcoustic) [reply fisherman-registry/src/test/java/fishermanregistry/collector/CollectorControllerIntegrationTest.java:46]: Okay, I created testing tools in a separate test.properties file and load them instead now.
About "Don't use toolSource in integration tests". How should I load the properties? Can I use my ToolProperties class, the one with ConfigurationProperties annotation? Or create a helper class exclusively for test tools?

Comment (piotrwestphal) [reply fisherman-registry/src/test/java/fishermanregistry/collector/CollectorControllerIntegrationTest.java:46]: Use the static value that matches value from the properties file :)

**Pull request merged**